felt/src/lib.rs

```diff
@@ -1052,6 +1088,204 @@ mod test {
+            let one = Felt::one();
+            prop_assert_eq!(&x, &(&x * &one));
+            prop_assert_eq!(&x, &(&one * &x));
+        }
+        #[test]
+        fn non_zero_felt_is_always_positive(ref x in FELT_NON_ZERO_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            prop_assert!(x.is_positive())
+        }
+
+        #[test]
+        fn felt_is_never_negative(ref x in FELT_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            prop_assert!(!x.is_negative())
+        }
+
+
+        #[test]
+        fn non_zero_felt_signum_is_always_one(ref x in FELT_NON_ZERO_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let one = Felt::one();
+            prop_assert_eq!(x.signum(), one)
+        }
+
+        #[test]
+        fn sub_abs(ref x in FELT_PATTERN, ref y in FELT_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let y = Felt::parse_bytes(y.as_bytes(), 10).unwrap();
+
+            let expected_abs_sub = if x > y {&x - &y} else {&y - &x};
+
+            prop_assert_eq!(&x.abs_sub(&y), &expected_abs_sub)
+        }
+
+        #[test]
+        fn abs(ref x in FELT_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            prop_assert_eq!(&x, &x.abs())
+        }
+
+        #[test]
+        fn modpow_in_range(ref x in FELT_PATTERN, ref y in FELT_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let y = &Felt::parse_bytes(y.as_bytes(), 10).unwrap();
+            let p = BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
+
+            let p_felt = Felt::max_value();
+
+            let modpow = x.modpow(y, &p_felt).to_biguint();
+            prop_assert!(modpow < p, "{}", modpow);
+        }
+
+        #[test]
+        fn sqrt_in_range(ref x in FELT_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let p = BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
+
+            let sqrt = x.sqrt().to_biguint();
+            prop_assert!(sqrt < p, "{}", sqrt);
+        }
+
+        #[test]
+        fn sqrt_is_inv_square(ref x in FELT_PATTERN) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            prop_assert_eq!((&x * &x).sqrt(), x);
+        }
+
+        #[test]
+        fn add_u32_in_range(ref x in FELT_PATTERN, y in any::<u32>()) {
+            let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let p = BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
+            let x_add_y = (x + y).to_biguint();
+            prop_assert!(x_add_y < p, "{}", x_add_y);
+        }
+
+        #[test]
+        fn add_u32_is_inv_sub(ref x in FELT_PATTERN, y in any::<u32>()) {
+            let x = &Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let expected_y = (x.clone() + y - x).to_u32().unwrap();
```

Comment:
`expected_y` does not necessarily fit in a `uint32`. `Felt(CAIRO_PRIME - 1) - u32::MAX` is still greater than `u32::MAX`.

Reply:
You should convert `y` to `BigUint` before the comparison instead.

Reply:
Why wouldn't it fit, we're subtracting `x` from `x`, we should be left with exactly `y` or am I missing something here ?

Reply:
Ugh, didn't see the `- x` 🤦